exporter/jaeger/lib/opentelemetry/exporter/jaeger/agent_exporter.rb

```diff
@@ -30,12 +30,14 @@ def initialize(host: ENV.fetch('OTEL_EXPORTER_JAEGER_AGENT_HOST', 'localhost'),
         # @param [Enumerable<OpenTelemetry::SDK::Trace::SpanData>] span_data the
         #   list of recorded {OpenTelemetry::SDK::Trace::SpanData} structs to be
         #   exported.
+        # @param [optional Numeric] timeout An optional timeout in seconds.
         # @return [Integer] the result of the export.
-        def export(span_data)
+        def export(span_data, timeout: nil)
           return FAILURE if @shutdown
 
+          start_time = Time.now
           encoded_batches(span_data) do |batch|
-            return FAILURE if @shutdown
+            return FAILURE if @shutdown || OpenTelemetry::SDK::Internal.maybe_timeout(timeout, start_time)&.zero?
```

Comment:
Not super awesome that we're reaching into an `Internal` SDK module here. Would we prefer duplication in each exporter gem?

Reply:
Is this something that could live in the soon to exist "utility gem"?  However simple the method may be, we probably want to keep it consistent across exporters right?  If not then I would just duplicate it into each gem.

Reply:
I think that’s a good place for it, yes.

Reply:
Should we pull this into `common`?

Reply:
Does it make sense to have the SDK depend on `common`?